Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/stats/EventAttributeValue.java

```diff
@@ -82,8 +82,30 @@ public EventAttributeValue(String value, IntArrayList occurCaseIndexes, long tot
             sum += act.getEventSize();
         }
         this.total = sum;
+
+        occurCasesIndexSet = occurActivities.stream()
+                .map(AActivity::getImmutableTraceIndex)
+                .collect(Collectors.toSet());
+    }
+
+    public EventAttributeValue(String value, UnifiedSet<Integer> occurCaseIndexes, long totalCases,
+                               UnifiedSet<AActivity> occurActivities) {
+        this.value = value.intern();
+        this.occurCasesIndexSet = occurCaseIndexes;
+        this.percent = 100 * ((double) occurCaseIndexes.size() / totalCases);
+        this.frequency = String.format("%.2f", percent );
```

Comment:
I think you can remove these, since they're now calculated on demand from occurCaseIndexSet in the .getPercent and .getFrequency methods.